lib/pbench/server/api/resources/users_api.py

```diff
@@ -1,542 +1,65 @@
 
-        user_payload = request.get_json()
-        if not user_payload:
-            current_app.logger.warning("Invalid json object: {}", request.url)
-            abort(HTTPStatus.BAD_REQUEST, message="Invalid json object in request")
-
-        result = self.get_valid_target_user(target_username, "PUT")
-        if not result.target_user:
-            abort(result.http_status, message=result.http_message)
-
-        # Check if the user payload contain fields that are either protected or
-        # are not present in the user db. If any key in the payload does not match
-        # with the column name we will abort the update request.
-        non_existent = set(user_payload.keys()).difference(
-            set(User.__table__.columns.keys())
-        )
-        if non_existent:
-            current_app.logger.warning(
-                "User trying to update fields that are not present in the user database. Fields: {}",
-                non_existent,
-            )
-            abort(
-                HTTPStatus.BAD_REQUEST,
-                message="Invalid fields in update request payload",
-            )
-        # Only admin user will be allowed to change other user's role. However,
-        # Admin users will not be able to change their admin role,
-        # This is done to prevent last admin user from de-admining him/herself
-        protected_db_fields = User.get_protected()
-        if (
-            not Auth.token_auth.current_user().is_admin()
-            or Auth.token_auth.current_user() == result.target_user
-        ):
-            protected_db_fields.append("role")
+        auth_token = Auth.get_auth_token()
+        current_user = Auth.token_auth.current_user()
+        response_object = {}
 
-        protected = set(user_payload.keys()).intersection(set(protected_db_fields))
-        for field in protected:
-            if getattr(result.target_user, field) != user_payload[field]:
-                current_app.logger.warning(
-                    "User trying to update the non-updatable fields. {}: {}",
-                    field,
-                    user_payload[field],
-                )
-                abort(HTTPStatus.FORBIDDEN, message="Invalid update request payload")
-        try:
-            result.target_user.update(**user_payload)
-        except Exception:
-            current_app.logger.exception(
-                "Exception occurred during updating user object"
-            )
-            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="INTERNAL ERROR")
+        if current_user.username == target_username:
+            token_payload = Auth.oidc_client.token_introspect(auth_token)
+            response_object["username"] = token_payload.get("preferred_username")
+            response_object["first_name"] = token_payload.get("given_name")
+            response_object["last_name"] = token_payload.get("family_name")
+            response_object["email"] = token_payload.get("email")
```

Comment:
This might be a bit easier to follow as,

```
response_object = {
    "username": token_payload.get("preferred_username"),
    "first_name": token_payload.get("given_name"),
    ...
}
```

which reads like JSON and therefore feels more consistent with the `elif`.